### ISSUE TITLE
False error reported when setting properties on a class which has a magic setter.

### ISSUE THREAD

**Issue by CowChris** [scrutinizer-ci/php-analyzer#666]
The ViewInterface mentioned in this error report has a magic method for setting properties:

```
    /**
     * Binds data to the view.
     *
     * @param string $key The key.
     * @param string $value The value.
     */
    public function __set($key, $value);
```

However, Scrutinizer penalizes us with a Major issue...

``` $view->categoryName = $selectedCategory->getName(); 

Bug introduced 3 days ago by Steve Heath  
Accessing categoryName on the interface Air\View\ViewInterface suggest that you code against a concrete implementation. How about adding an instanceof check?

```

Would not expect it to raise an issue.

see https://scrutinizer-ci.com/b/easyfundraising/easyfundraising/inspections/af77792a-0ab4-4099-bbe1-1568e6bda664/issues/files/app/Block/Navigation/Categories/Controller/Categories.php?status=new&orderField=path&order=asc&issueId=029d3d36-cf25-4ff7-9825-1ee0715f8040
